cmd/manager/commands/cluster.go

```diff
@@ -0,0 +1,156 @@
+package commands
+
+import (
+	"context"
+	"fmt"
+	"github.com/keikoproj/manager/internal/config/common"
+	"github.com/keikoproj/manager/internal/utils"
+	"github.com/keikoproj/manager/pkg/k8s"
+	"github.com/spf13/cobra"
+	"k8s.io/api/rbac/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	"log"
+	"os"
+)
+
+// NewClusterCommand returns a new instance of an `manager cluster` command
+func NewClusterCommand() *cobra.Command {
+	var command = &cobra.Command{
+		Use:   "cluster",
+		Short: "Manage cluster operations",
+		Run: func(c *cobra.Command, args []string) {
+			c.HelpFunc()(c, args)
+			os.Exit(1)
+		},
+		Example: `  # Register a cluster with keiko-manager. The context must exist in your kubectl config:
+  manager cluster register -ctx admins@iksm-ppd-usw2-k8s
+  #	Remove managed cluster from manager
+  manager cluster unregister -c admins@iksm-ppd-usw2-k8s
+`,
+	}
+
+	command.AddCommand(NewClusterRegisterCommand())
+	command.AddCommand(NewClusterUnregisterCommand())
+	return command
+}
+
+//NewClusterRegisterCommand registers the target cluster with the manager.
+//Target cluster can be the same cluster where manager resides (provide --self true)
+//If service account is not provided manager will create service account, cluster role and role binding
+//Service Account must exists in "kube-system" namespace
+func NewClusterRegisterCommand() *cobra.Command {
+	var (
+		self           bool
+		serviceAccount string
+		configContext  string
+	)
+
+	var command = &cobra.Command{
+		Use:     "register",
+		Short:   fmt.Sprintf("%s cluster register", "manager"),
+		Long:    "Add/register managed cluster with manager",
+		Example: "manager cluster register -c admins@iksm-ppd-usw2-k8s",
+		Run: func(c *cobra.Command, args []string) {
+			ctx := context.Background()
+			clientSet := getManagedClusterKubeConfig(configContext)
+			managedClusterClient := k8s.NewK8sManagedClusterClientDoOrDie(clientSet)
+			if serviceAccount == "" {
+				createRBACInManagedCluster(ctx, managedClusterClient)
+				serviceAccount = common.ManagerServiceAccountName
+			}
+			_, err := managedClusterClient.GetServiceAccountTokenSecret(ctx, serviceAccount, common.SystemNameSpace)
+			utils.StopIfError(err)
+			fmt.Printf("token received successfully\n")
+		},
+	}
+
+	command.Flags().BoolVarP(&self, "self", "i", false, "To self manage keiko manager cluster itself. Default = false")
```

Comment:
Does this mean to register central cluster to itself? what's the meaning here, since central cluster should be able to access itself through kubeconfig.

Reply:
How do you get that kubeconfig? This command is how we get the kubeconfig file but for self clusters we don't need API server ALB url instead we can use "kubernetes.default.svc" so it doesn't go out. 

i will have to take care of it though before sending that cluster information to server